### PR TITLE
.cirlceci: Keep tags that look like a sha1

### PR DESCRIPTION
**Pull request by seemethere**
Previous attempts to get this right:
* https://github.com/pytorch/pytorch/pull/38335
* https://github.com/pytorch/pytorch/pull/38279
* https://github.com/pytorch/pytorch/pull/37976

This tag kept getting deleted before the docker image ci workflow could
be merged causing it to have upstream breakages.

It'd be best to make sure the garbage collector just doesnt garbage
collect it.

This is a pre-step to merge https://github.com/pytorch/pytorch/pull/38484

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

### CODE REVIEW

**Review by malfet — APPROVED** [review]
I trust you with those numbers

**Comment by seemethere** [comment]
@malfet Rewrote the script after a chat with @ezyang to instead look for tags that look like a sha1 and then add them to the "stable" time window instead.

This should solve our use case going forward.

Here's a test run that I did: https://hastebin.com/bamefobufi.pl

File: .circleci/ecr_gc_docker/gc.py [hunk 1]
```diff
@@ -146,6 +147,17 @@ def chunks(chunkable, n):
...
 
+SHA_PATTERN = re.compile(r'\b[0-9a-f]{40}\b')
+def looks_like_git_sha(tag):
+    """Returns a boolean to check if a tag looks like a git sha
+
+    For reference a sha1 is 40 characters with only 0-9a-f and contains no
+    "-" characters
+    """
+    return all([
+        re.match(SHA_PATTERN, tag) is not None,
```
Comment (ezyang) [inline .circleci/ecr_gc_docker/gc.py:158]: Why don't you just do a full match (aka `^PATTERN$`) here?

Comment (seemethere) [reply .circleci/ecr_gc_docker/gc.py]: Updated to do full match

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**